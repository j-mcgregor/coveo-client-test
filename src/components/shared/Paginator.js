import React from 'react';
import PropTypes from 'prop-types';
import Paginator from 'react-hooks-paginator';
import S from '../../static/styles';

const PaginatorWrapper = ({
  total,
  pageLimit,
  setOffset,
  currentPage,
  setCurrentPage
}) => {
  return (
    <S.Paginator>
      <Paginator
        totalRecords={total}
        pageLimit={pageLimit}
        pageNeighbours={2}
        setOffset={setOffset}
        currentPage={currentPage}
        setCurrentPage={setCurrentPage}
        pagePrevText="« Prev"
        pageNextText="Next »"
      />
    </S.Paginator>
  );
};

PaginatorWrapper.propTypes = {
  total: PropTypes.number,
  pageLimit: PropTypes.number,
  setOffset: PropTypes.func,
  currentPage: PropTypes.number,
  setCurrentPage: PropTypes.func
};

PaginatorWrapper.defaultProps = {
  total: null,
  pageLimit: 12,
  setOffset: () => {},
  currentPage: 1,
  setCurrentPage: () => {}
};

export default PaginatorWrapper;
