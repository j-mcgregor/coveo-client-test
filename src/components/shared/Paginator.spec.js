import React from 'react';
import { shallow } from 'enzyme';
import Paginator from './Paginator';

describe('Paginator', () => {
  it('should render the Paginator', () => {
    const props = {
      total: 100,
      pageLimit: 10,
      setOffset: () => {},
      currentPage: 1,
      setCurrentPage: () => {}
    };

    const wrapper = shallow(<Paginator {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
