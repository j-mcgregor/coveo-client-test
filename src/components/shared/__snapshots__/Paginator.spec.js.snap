// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator should render the Paginator 1`] = `ShallowWrapper {}`;
